src/modules/iKinGazeCtrl/src/main.cpp

```diff
@@ -1225,18 +1233,38 @@ class GazeModule: public RFModule
         imuPort.setExchangeData(&commData);
         if (commData.stabilizationOn)
         {
-            imuPort.open(commData.localStemName+"/inertial:i");
-            if (Network::connect(remoteInertialName,imuPort.getName()))
-                yInfo("Receiving IMU data from %s",remoteInertialName.c_str());
-            else
-            {
-                yError("Unable to connect to %s!",remoteInertialName.c_str());
-                dispose();
-                return false;
+            if (commData.useMASClient) {
+                Property mas_conf{{"device", Value("multipleanalogsensorsclient")},
+                                  {"remote", Value("/"+commData.robotName+"/head/imurfe")},
+                                  {"local",  Value("/"+commData.localStemName+"/head/imurfe:i")}};
+
+                if (!(mas_client.open(mas_conf)) ||
```

Comment:
I personally don't like this, because we compact checks that shall be sequential in one single verification relying on the properties of the compiler (and hiding this detail). Better off sticking to two subsequent checks instead.